File: Search/ECMSearchWPF/CS/app_code/clsDATASOURCERESTOREHISTORY.cs
// VBConversions Note: VB project level imports
using System.Windows.Input;
using System.Windows.Data;
using System.Windows.Documents;
using System.Xml.Linq;
using System.Collections.Generic;
using System.Windows.Navigation;
using System.Windows.Media.Imaging;
using System.Windows;
using Microsoft.VisualBasic;
using System.Windows.Media;
using System.Collections;
using System;
using System.Windows.Shapes;
using System.Windows.Controls;
using System.Threading.Tasks;
using System.Linq;
using System.Diagnostics;
// End of VB project level imports

using System.Configuration;
using System.IO;

//Imports System.Data.SqlClient
//Imports System.Data.Sql

namespace ECMSearchWPF
{
	public class clsDATASOURCERESTOREHISTORY
	{
		
		clsCommonFunctions COMMON = new clsCommonFunctions();
		//Dim EP As New clsEndPoint
		clsEncryptV2 ENC2 = new clsEncryptV2();
		
		
		//Dim proxy As New SVCSearch.Service1Client
		
		//** DIM the selected table columns
		//Dim DB As New clsDatabase
		clsDma DMA = new clsDma();
		clsUtility UTIL = new clsUtility();
		clsLogging LOG = new clsLogging();
		
		
		string SourceGuid = "";
		string RestoredToMachine = "";
		string RestoreUserName = "";
		string RestoreUserID = "";
		string RestoreUserDomain = "";
		string RestoreDate = "";
		string DataSourceOwnerUserID = "";
		string SeqNo = "";
		string TypeContentCode = "";
		string CreateDate = "";
		string DocumentName = "";
		string FQN = "";
		string VerifiedData = "";
		string OrigCrc = "";
		string RestoreCrc = "";
		
		bool HandlerAdded = false;
		public clsDATASOURCERESTOREHISTORY()
		{
			
		}
		//** Generate the SET methods
		public void setSourceguid(ref string val)
		{
			if (val.Length == 0)
			{
				MessageBox.Show("clsDATASOURCERESTOREHISTORY - SET: Field \'Sourceguid\' cannot be NULL.");
				return;
			}
			val = UTIL.RemoveSingleQuotes(val);
			SourceGuid = val;
		}
		
		public void setRestoredtomachine(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			RestoredToMachine = val;
		}
		
		public void setRestoreusername(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			RestoreUserName = val;
		}
		
		public void setRestoreuserid(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			RestoreUserID = val;
		}
		
		public void setRestoreuserdomain(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			RestoreUserDomain = val;
		}
		
		public void setRestoredate(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			RestoreDate = val;
		}
		
		public void setDatasourceowneruserid(ref string val)
		{
			if (val.Length == 0)
			{
				MessageBox.Show("clsDATASOURCERESTOREHISTORY - SET: Field \'Datasourceowneruserid\' cannot be NULL.");
				return;
			}
			val = UTIL.RemoveSingleQuotes(val);
			DataSourceOwnerUserID = val;
		}
		
		public void setTypecontentcode(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			TypeContentCode = val;
		}
		
		public void setCreatedate(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			CreateDate = val;
		}
		
		public void setDocumentname(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			DocumentName = val;
		}
		
		public void setFqn(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			FQN = val;
		}
		
		public void setVerifieddata(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			VerifiedData = val;
		}
		
		public void setOrigcrc(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			OrigCrc = val;
		}
		
		public void setRestorecrc(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			RestoreCrc = val;
		}
		
		
		
		//** Generate the GET methods
		public string getSourceguid()
		{
			if (SourceGuid.Length == 0)
			{
				MessageBox.Show("clsDATASOURCERESTOREHISTORY - GET: Field \'Sourceguid\' cannot be NULL.");
				return "";
			}
			return UTIL.RemoveSingleQuotes(SourceGuid);
		}
		
		public string getRestoredtomachine()
		{
			return UTIL.RemoveSingleQuotes(RestoredToMachine);
		}
		
		public string getRestoreusername()
		{
			return UTIL.RemoveSingleQuotes(RestoreUserName);
		}
		
		public string getRestoreuserid()
		{
			return UTIL.RemoveSingleQuotes(RestoreUserID);
		}
		
		public string getRestoreuserdomain()
		{
			return UTIL.RemoveSingleQuotes(RestoreUserDomain);
		}
		
		public string getRestoredate()
		{
			return UTIL.RemoveSingleQuotes(RestoreDate);
		}
		
		public string getDatasourceowneruserid()
		{
			if (DataSourceOwnerUserID.Length == 0)
			{
				MessageBox.Show("clsDATASOURCERESTOREHISTORY - GET: Field \'Datasourceowneruserid\' cannot be NULL.");
				return "";
			}
			return UTIL.RemoveSingleQuotes(DataSourceOwnerUserID);
		}
		
		public string getSeqno()
		{
			if (SeqNo.Length == 0)
			{
				MessageBox.Show("clsDATASOURCERESTOREHISTORY - GET: Field \'Seqno\' cannot be NULL.");
				return "";
			}
			if (SeqNo.Length == 0)
			{
				SeqNo = "null";
			}
			return SeqNo;
		}
		
		public string getTypecontentcode()
		{
			return UTIL.RemoveSingleQuotes(TypeContentCode);
		}
		
		public string getCreatedate()
		{
			return UTIL.RemoveSingleQuotes(CreateDate);
		}
		
		public string getDocumentname()
		{
			return UTIL.RemoveSingleQuotes(DocumentName);
		}
		
		public string getFqn()
		{
			return UTIL.RemoveSingleQuotes(FQN);
		}
		
		public string getVerifieddata()
		{
			return UTIL.RemoveSingleQuotes(VerifiedData);
		}
		
		public string getOrigcrc()
		{
			return UTIL.RemoveSingleQuotes(OrigCrc);
		}
		
		public string getRestorecrc()
		{
			return UTIL.RemoveSingleQuotes(RestoreCrc);
		}
		
		
		
		//** Generate the Required Fields Validation method
		public bool ValidateReqData()
		{
			if (SourceGuid.Length == 0)
			{
				return false;
			}
			if (DataSourceOwnerUserID.Length == 0)
			{
				return false;
			}
			if (SeqNo.Length == 0)
			{
				return false;
			}
			return true;
		}
		
		
		//** Generate the Validation method
		public bool ValidateData()
		{
			if (SourceGuid.Length == 0)
			{
				return false;
			}
			if (DataSourceOwnerUserID.Length == 0)
			{
				return false;
			}
			return true;
		}
		
		
		//** Generate the INSERT method
		public bool Insert()
		{
			bool b = false;
			string s = "";
			s = s + " INSERT INTO DataSourceRestoreHistory(" + "\r\n";
			s = s + "SourceGuid," + "\r\n";
			s = s + "RestoredToMachine," + "\r\n";
			s = s + "RestoreUserName," + "\r\n";
			s = s + "RestoreUserID," + "\r\n";
			s = s + "RestoreUserDomain," + "\r\n";
			s = s + "RestoreDate," + "\r\n";
			s = s + "DataSourceOwnerUserID," + "\r\n";
			s = s + "TypeContentCode," + "\r\n";
			s = s + "CreateDate)" + "\r\n";
			s = s + " values (" + "\r\n";
			s = s + "\'" + SourceGuid + "\'" + "," + "\r\n";
			s = s + "\'" + RestoredToMachine + "\'" + "," + "\r\n";
			s = s + "\'" + RestoreUserName + "\'" + "," + "\r\n";
			s = s + "\'" + RestoreUserID + "\'" + "," + "\r\n";
			s = s + "\'" + RestoreUserDomain + "\'" + "," + "\r\n";
			s = s + "\'" + DateTime.Now.ToString() + "\'" + "," + "\r\n";
			s = s + "\'" + DataSourceOwnerUserID + "\'" + "," + "\r\n";
			s = s + "\'" + TypeContentCode + "\'" + "," + "\r\n";
			s = s + "\'" + DateTime.Now.ToString() + "\')" + "\r\n";
			
			if (! HandlerAdded)
			{
				GLOBALS.ProxySearch.ExecuteSqlNewConnSecureCompleted += new System.EventHandler(client_ExecuteSqlNewConn);
				//EP.setSearchSvcEndPoint(proxy)
			}
			s = ENC2.EncryptPhrase(s, GLOBALS.ContractID);
			GLOBALS.ProxySearch.ExecuteSqlNewConnSecureAsync(GLOBALS._SecureID, GLOBALS.gRowID, s, GLOBALS._UserID, GLOBALS.ContractID);
			
			return true;
			
		}
		
		
		//** Generate the UPDATE method
		public bool Update(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			s = s + " update DataSourceRestoreHistory set ";
			s = s + "SourceGuid = \'" + getSourceguid() + "\'" + ", ";
			s = s + "RestoredToMachine = \'" + getRestoredtomachine() + "\'" + ", ";
			s = s + "RestoreUserName = \'" + getRestoreusername() + "\'" + ", ";
			s = s + "RestoreUserID = \'" + getRestoreuserid() + "\'" + ", ";
			s = s + "RestoreUserDomain = \'" + getRestoreuserdomain() + "\'" + ", ";
			s = s + "RestoreDate = \'" + getRestoredate() + "\'" + ", ";
			s = s + "DataSourceOwnerUserID = \'" + getDatasourceowneruserid() + "\'" + ", ";
			s = s + "TypeContentCode = \'" + getTypecontentcode() + "\'" + ", ";
			s = s + "CreateDate = \'" + getCreatedate() + "\'" + ", ";
			s = s + "DocumentName = \'" + getDocumentname() + "\'" + ", ";
			s = s + "FQN = \'" + getFqn() + "\'" + ", ";
			s = s + "VerifiedData = \'" + getVerifieddata() + "\'" + ", ";
			s = s + "OrigCrc = \'" + getOrigcrc() + "\'" + ", ";
			s = s + "RestoreCrc = \'" + getRestorecrc() + "\'";
			WhereClause = (string) (" " + WhereClause);
			s = s + WhereClause;
			if (! HandlerAdded)
			{
				GLOBALS.ProxySearch.ExecuteSqlNewConnSecureCompleted += new System.EventHandler(client_ExecuteSqlNewConn);
				//EP.setSearchSvcEndPoint(proxy)
			}
			
			s = ENC2.EncryptPhrase(s, GLOBALS.ContractID);
			GLOBALS.ProxySearch.ExecuteSqlNewConnSecureAsync(GLOBALS._SecureID, GLOBALS.gRowID, s, GLOBALS._UserID, GLOBALS.ContractID);
			
			return true;
		}
		
		
		
		
		//** Generate the DELETE method
		public bool Delete(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			WhereClause = (string) (" " + WhereClause);
			
			s = " Delete from DataSourceRestoreHistory";
			s = s + WhereClause;
			
			if (! HandlerAdded)
			{
				GLOBALS.ProxySearch.ExecuteSqlNewConnSecureCompleted += new System.EventHandler(client_ExecuteSqlNewConn);
				//EP.setSearchSvcEndPoint(proxy)
			}
			
			s = ENC2.EncryptPhrase(s, GLOBALS.ContractID);
			GLOBALS.ProxySearch.ExecuteSqlNewConnSecureAsync(GLOBALS._SecureID, GLOBALS.gRowID, s, GLOBALS._UserID, GLOBALS.ContractID);
			
			return true;
			
		}
		
		
		//** Generate the Zeroize Table method
		public bool Zeroize()
		{
			bool b = false;
			string s = "";
			
			s = s + " Delete from DataSourceRestoreHistory";
			
			if (! HandlerAdded)
			{
				GLOBALS.ProxySearch.ExecuteSqlNewConnSecureCompleted += new System.EventHandler(client_ExecuteSqlNewConn);
				//EP.setSearchSvcEndPoint(proxy)
			}
			
			s = ENC2.EncryptPhrase(s, GLOBALS.ContractID);
			GLOBALS.ProxySearch.ExecuteSqlNewConnSecureAsync(GLOBALS._SecureID, GLOBALS.gRowID, s, GLOBALS._UserID, GLOBALS.ContractID);
			
			return true;
			
		}
		
		
		//'** Generate Index Queries
		//Public Function cnt_PI01_RestoreHist(ByVal DataSourceOwnerUserID As String, ByVal TypeContentCode As String, ByVal VerifiedData As String) As Integer
		
		//    Dim B As Integer = 0
		//    Dim TBL$ = "DataSourceRestoreHistory"
		//    Dim WC$ = "Where DataSourceOwnerUserID = '" + DataSourceOwnerUserID + "' and   TypeContentCode = '" + TypeContentCode + "' and   VerifiedData = '" + VerifiedData + "'"
		
		//    B = DB.iGetRowCount(TBL$, WC)
		
		//    Return B
		//End Function     '** cnt_PI01_RestoreHist
		//Public Function cnt_PK83(ByVal SeqNo As Integer) As Integer
		
		//    Dim B As Integer = 0
		//    Dim TBL$ = "DataSourceRestoreHistory"
		//    Dim WC$ = "Where SeqNo = " & SeqNo
		
		//    B = DB.iGetRowCount(TBL$, WC)
		
		//    Return B
		//End Function     '** cnt_PK83
		
		//'** Generate Index ROW Queries
		//Public Function getRow_PI01_RestoreHist(ByVal DataSourceOwnerUserID As String, ByVal TypeContentCode As String, ByVal VerifiedData As String) As SqlDataReader
		
		//    Dim rsData As SqlDataReader = Nothing
		//    Dim TBL$ = "DataSourceRestoreHistory"
		//    Dim WC$ = "Where DataSourceOwnerUserID = '" + DataSourceOwnerUserID + "' and   TypeContentCode = '" + TypeContentCode + "' and   VerifiedData = '" + VerifiedData + "'"
		
		//    rsData = DB.GetRowByKey(TBL$, WC)
		
		//    If rsData.HasRows Then
		//        Return rsData
		//    Else
		//        Return Nothing
		//    End If
		//End Function     '** getRow_PI01_RestoreHist
		//Public Function getRow_PK83(ByVal SeqNo As Integer) As SqlDataReader
		
		//    Dim rsData As SqlDataReader = Nothing
		//    Dim TBL$ = "DataSourceRestoreHistory"
		//    Dim WC$ = "Where SeqNo = " & SeqNo
		
		//    rsData = DB.GetRowByKey(TBL$, WC)
		
		//    If rsData.HasRows Then
		//        Return rsData
		//    Else
		//        Return Nothing
		//    End If
		//End Function     '** getRow_PK83
		
		// ''' Build Index Where Caluses
		// '''
		//Public Function wc_PI01_RestoreHist(ByVal DataSourceOwnerUserID As String, ByVal TypeContentCode As String, ByVal VerifiedData As String) As String
		
		//    Dim WC$ = "Where DataSourceOwnerUserID = '" + DataSourceOwnerUserID + "' and   TypeContentCode = '" + TypeContentCode + "' and   VerifiedData = '" + VerifiedData + "'"
		
		//    Return WC
		//End Function     '** wc_PI01_RestoreHist
		//Public Function wc_PK83(ByVal SeqNo As Integer) As String
		
		//    Dim WC$ = "Where SeqNo = " & SeqNo
		
		//    Return WC
		//End Function     '** wc_PK83
		
		public void client_ExecuteSqlNewConn(object sender, SVCSearch.ExecuteSqlNewConnSecureCompletedEventArgs e)
		{
			int TrackingNbr = 0;
			if (e.Error == null)
			{
				bool B = System.Convert.ToBoolean(e.Result);
				if (B)
				{
					
				}
			}
			else
			{
				modGlobals.gErrorCount++;
				LOG.WriteToSqlLog((string) ("ERROR clsSql 100: " + e.Error.Message));
			}
		}
		~clsDATASOURCERESTOREHISTORY()
		{
			try
			{
				
			}
			finally
			{
				base.Finalize(); //define the destructor
				HandlerAdded = false;
				GLOBALS.ProxySearch.ExecuteSqlNewConnSecureCompleted -= new System.EventHandler(client_ExecuteSqlNewConn);
				GC.Collect();
				GC.WaitForPendingFinalizers();
			}
		}
		
		
	}
	
}

File: Search/ECMSearchWPF/app_code/clsDATASOURCERESTOREHISTORY.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
'Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports ECMEncryption

Public Class clsDATASOURCERESTOREHISTORY

    Dim COMMON As New clsCommonFunctions
    'Dim EP As New clsEndPoint
    Dim ENC2 As New ECMEncrypt()


    'Dim proxy As New SVCSearch.Service1Client

    '** DIM the selected table columns 
    'Dim DB As New clsDatabase
    Dim DMA As New clsDma
    Dim UTIL As New clsUtility
    Dim LOG As New clsLogMain


    Dim SourceGuid As String = ""
    Dim RestoredToMachine As String = ""
    Dim RestoreUserName As String = ""
    Dim RestoreUserID As String = ""
    Dim RestoreUserDomain As String = ""
    Dim RestoreDate As String = ""
    Dim DataSourceOwnerUserID As String = ""
    Dim SeqNo As String = ""
    Dim TypeContentCode As String = ""
    Dim CreateDate As String = ""
    Dim DocumentName As String = ""
    Dim FQN As String = ""
    Dim VerifiedData As String = ""
    Dim OrigCrc As String = ""
    Dim RestoreCrc As String = ""

    Dim HandlerAdded As Boolean = False
    Sub New()

    End Sub
    '** Generate the SET methods 
    Public Sub setSourceguid(ByRef val As String)
        If Len(val) = 0 Then
            messagebox.show("clsDATASOURCERESTOREHISTORY - SET: Field 'Sourceguid' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        SourceGuid = val
    End Sub

    Public Sub setRestoredtomachine(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        RestoredToMachine = val
    End Sub

    Public Sub setRestoreusername(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        RestoreUserName = val
    End Sub

    Public Sub setRestoreuserid(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        RestoreUserID = val
    End Sub

    Public Sub setRestoreuserdomain(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        RestoreUserDomain = val
    End Sub

    Public Sub setRestoredate(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        RestoreDate = val
    End Sub

    Public Sub setDatasourceowneruserid(ByRef val As String)
        If Len(val) = 0 Then
            messagebox.show("clsDATASOURCERESTOREHISTORY - SET: Field 'Datasourceowneruserid' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        DataSourceOwnerUserID = val
    End Sub

    Public Sub setTypecontentcode(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        TypeContentCode = val
    End Sub

    Public Sub setCreatedate(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        CreateDate = val
    End Sub

    Public Sub setDocumentname(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        DocumentName = val
    End Sub

    Public Sub setFqn(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        FQN = val
    End Sub

    Public Sub setVerifieddata(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        VerifiedData = val
    End Sub

    Public Sub setOrigcrc(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        OrigCrc = val
    End Sub

    Public Sub setRestorecrc(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        RestoreCrc = val
    End Sub



    '** Generate the GET methods 
    Public Function getSourceguid() As String
        If Len(SourceGuid) = 0 Then
            messagebox.show("clsDATASOURCERESTOREHISTORY - GET: Field 'Sourceguid' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(SourceGuid)
    End Function

    Public Function getRestoredtomachine() As String
        Return UTIL.RemoveSingleQuotes(RestoredToMachine)
    End Function

    Public Function getRestoreusername() As String
        Return UTIL.RemoveSingleQuotes(RestoreUserName)
    End Function

    Public Function getRestoreuserid() As String
        Return UTIL.RemoveSingleQuotes(RestoreUserID)
    End Function

    Public Function getRestoreuserdomain() As String
        Return UTIL.RemoveSingleQuotes(RestoreUserDomain)
    End Function

    Public Function getRestoredate() As String
        Return UTIL.RemoveSingleQuotes(RestoreDate)
    End Function

    Public Function getDatasourceowneruserid() As String
        If Len(DataSourceOwnerUserID) = 0 Then
            messagebox.show("clsDATASOURCERESTOREHISTORY - GET: Field 'Datasourceowneruserid' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(DataSourceOwnerUserID)
    End Function

    Public Function getSeqno() As String
        If Len(SeqNo) = 0 Then
            messagebox.show("clsDATASOURCERESTOREHISTORY - GET: Field 'Seqno' cannot be NULL.")
            Return ""
        End If
        If Len(SeqNo) = 0 Then
            SeqNo = "null"
        End If
        Return SeqNo
    End Function

    Public Function getTypecontentcode() As String
        Return UTIL.RemoveSingleQuotes(TypeContentCode)
    End Function

    Public Function getCreatedate() As String
        Return UTIL.RemoveSingleQuotes(CreateDate)
    End Function

    Public Function getDocumentname() As String
        Return UTIL.RemoveSingleQuotes(DocumentName)
    End Function

    Public Function getFqn() As String
        Return UTIL.RemoveSingleQuotes(FQN)
    End Function

    Public Function getVerifieddata() As String
        Return UTIL.RemoveSingleQuotes(VerifiedData)
    End Function

    Public Function getOrigcrc() As String
        Return UTIL.RemoveSingleQuotes(OrigCrc)
    End Function

    Public Function getRestorecrc() As String
        Return UTIL.RemoveSingleQuotes(RestoreCrc)
    End Function



    '** Generate the Required Fields Validation method 
    Public Function ValidateReqData() As Boolean
        If SourceGuid.Length = 0 Then Return False
        If DataSourceOwnerUserID.Length = 0 Then Return False
        If SeqNo.Length = 0 Then Return False
        Return True
    End Function


    '** Generate the Validation method 
    Public Function ValidateData() As Boolean
        If SourceGuid.Length = 0 Then Return False
        If DataSourceOwnerUserID.Length = 0 Then Return False
        Return True
    End Function


    '** Generate the INSERT method 
    Public Function Insert() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""
        s = s + " INSERT INTO DataSourceRestoreHistory(" + Environment.NewLine
        s = s + "SourceGuid," + Environment.NewLine
        s = s + "RestoredToMachine," + Environment.NewLine
        s = s + "RestoreUserName," + Environment.NewLine
        s = s + "RestoreUserID," + Environment.NewLine
        s = s + "RestoreUserDomain," + Environment.NewLine
        s = s + "RestoreDate," + Environment.NewLine
        s = s + "DataSourceOwnerUserID," + Environment.NewLine
        s = s + "TypeContentCode," + Environment.NewLine
        s = s + "CreateDate)" + Environment.NewLine
        s = s + " values (" + Environment.NewLine
        s = s + "'" + SourceGuid + "'" + "," + Environment.NewLine
        s = s + "'" + RestoredToMachine + "'" + "," + Environment.NewLine
        s = s + "'" + RestoreUserName + "'" + "," + Environment.NewLine
        s = s + "'" + RestoreUserID + "'" + "," + Environment.NewLine
        s = s + "'" + RestoreUserDomain + "'" + "," + Environment.NewLine
        s = s + "'" + Now.ToString + "'" + "," + Environment.NewLine
        s = s + "'" + DataSourceOwnerUserID + "'" + "," + Environment.NewLine
        s = s + "'" + TypeContentCode + "'" + "," + Environment.NewLine
        s = s + "'" + Now.ToString + "')" + Environment.NewLine

        If Not HandlerAdded Then
            ''AddHandler ProxySearch.ExecuteSqlNewConnSecureCompleted, AddressOf gLogSQL
            'EP.setSearchSvcEndPoint(proxy)
        End If
        s = ENC2.AES256EncryptString(s)
        Dim rc As Boolean = ProxySearch.ExecuteSqlNewConnSecure(_SecureID, s, _UserID, ContractID)
        gLogSQL(rc, ENC2.AES256EncryptString(s))

        Return True

    End Function


    '** Generate the UPDATE method 
    Public Function Update(ByVal WhereClause As String) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        s = s + " update DataSourceRestoreHistory set "
        s = s + "SourceGuid = '" + getSourceguid() + "'" + ", "
        s = s + "RestoredToMachine = '" + getRestoredtomachine() + "'" + ", "
        s = s + "RestoreUserName = '" + getRestoreusername() + "'" + ", "
        s = s + "RestoreUserID = '" + getRestoreuserid() + "'" + ", "
        s = s + "RestoreUserDomain = '" + getRestoreuserdomain() + "'" + ", "
        s = s + "RestoreDate = '" + getRestoredate() + "'" + ", "
        s = s + "DataSourceOwnerUserID = '" + getDatasourceowneruserid() + "'" + ", "
        s = s + "TypeContentCode = '" + getTypecontentcode() + "'" + ", "
        s = s + "CreateDate = '" + getCreatedate() + "'" + ", "
        s = s + "DocumentName = '" + getDocumentname() + "'" + ", "
        s = s + "FQN = '" + getFqn() + "'" + ", "
        s = s + "VerifiedData = '" + getVerifieddata() + "'" + ", "
        s = s + "OrigCrc = '" + getOrigcrc() + "'" + ", "
        s = s + "RestoreCrc = '" + getRestorecrc() + "'"
        WhereClause = " " + WhereClause
        s = s + WhereClause
        If Not HandlerAdded Then
            ''AddHandler ProxySearch.ExecuteSqlNewConnSecureCompleted, AddressOf gLogSQL
            'EP.setSearchSvcEndPoint(proxy)
        End If

        s = ENC2.AES256EncryptString(s)
        Dim BB As Boolean = ProxySearch.ExecuteSqlNewConnSecure(_SecureID, s, _UserID, ContractID)
        gLogSQL(BB, ENC2.AES256EncryptString(s))
        Return True
    End Function




    '** Generate the DELETE method 
    Public Function Delete(ByVal WhereClause As String) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        WhereClause = " " + WhereClause

        s = " Delete from DataSourceRestoreHistory"
        s = s + WhereClause

        If Not HandlerAdded Then
            ''AddHandler ProxySearch.ExecuteSqlNewConnSecureCompleted, AddressOf gLogSQL
            'EP.setSearchSvcEndPoint(proxy)
        End If

        s = ENC2.AES256EncryptString(s)
        Dim bb As Boolean = ProxySearch.ExecuteSqlNewConnSecure(_SecureID, s, _UserID, ContractID)
        gLogSQL(bb, ENC2.AES256EncryptString(s))

        Return True

    End Function


    '** Generate the Zeroize Table method 
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        s = s + " Delete from DataSourceRestoreHistory"

        If Not HandlerAdded Then
            ''AddHandler ProxySearch.ExecuteSqlNewConnSecureCompleted, AddressOf gLogSQL
            'EP.setSearchSvcEndPoint(proxy)
        End If

        s = ENC2.AES256EncryptString(s)
        Dim BB As Boolean = ProxySearch.ExecuteSqlNewConnSecure(_SecureID, s, _UserID, ContractID)
        gLogSQL(BB, ENC2.AES256EncryptString(s))
        Return True

    End Function



    Protected Overrides Sub Finalize()
        Try

        Finally
            MyBase.Finalize()      'define the destructor
            HandlerAdded = False
            ''RemoveHandler ProxySearch.ExecuteSqlNewConnSecureCompleted, AddressOf gLogSQL
            GC.Collect()
            GC.WaitForPendingFinalizers()
        End Try
    End Sub


End Class
